core/src/main/resources/xsl/raxRoles.xsl

```diff
@@ -113,15 +113,24 @@
 
     <xsl:template name="generateRoles">
         <xsl:param name="roles" as="xsd:string*" select="()"/>
-        <xsl:if test="not('#all' = $roles)">
-            <xsl:for-each select="$roles">
-                <wadl:param name="X-ROLES" style="header" rax:code="403" rax:anyMatch="true"
-                            rax:message="You are forbidden to perform the operation" type="xsd:string"
-                            required="true" repeating="true">
-                    <xsl:attribute name="fixed" select="rax:transformNBSP(.)"/>
-                </wadl:param>
-            </xsl:for-each>
-        </xsl:if>
+        <xsl:choose>
+            <xsl:when test="not('#all' = $roles)">
+                <rax:captureHeader name="X-RELEVANT-ROLES">
+                    <xsl:attribute name="path" select="string-join(('for $h in req:headers(''X-ROLES'', true()) return if ($h = (''', string-join($roles, ''', '''), ''')) then $h else ()'), '')"/>
```

Comment:
I'm really not very happy with this. I'm sure there is a better way to express the same thing, but I only know enough XPath to be dangerous.

The thing to point out here is that the `$roles` param is not available in the XPath context, so it can't be inserted directly into the `path` attribute. Instead, `xsl:attribute` is used to populate the `path` attribute. Basically, I'm just using the context that knows about `$roles` to construct a string that will become the value of the `path` attribute. The value of the `path` attribute is an XPath expression which will compare the values from the `X-Roles` header with the values of configured `rax:roles`. Note that the value of the `path` attribute is an XPath expression that will be evaluated in a context which provides the `req:headers` functions.